### PR TITLE
Fix/gnc optimizer

### PR DESCRIPTION
**Pull request by lucacarlone**
- [x] fixed corner cases in which mu = 0 or weights become negative
- [x] improved verbosity handling
- [x] added clone function to support PoseToPointFactor

### CODE REVIEW

File: gtsam/geometry/triangulation.h [hunk 1]
```diff
@@ -190,13 +193,14 @@ Point3 triangulateNonlinear(const std::vector<Pose3>& poses,
 template<class CAMERA>
 Point3 triangulateNonlinear(
     const CameraSet<CAMERA>& cameras,
-    const typename CAMERA::MeasurementVector& measurements, const Point3& initialEstimate) {
+    const typename CAMERA::MeasurementVector& measurements, const Point3& initialEstimate,
+    const SharedNoiseModel& model = nullptr) {
```
Comment (varunagrawal) [inline gtsam/geometry/triangulation.h:197]: Even these top-level methods can take the same default maybe?

**Review by varunagrawal — APPROVED** [review]
Two small comments but otherwise LGTM!

File: gtsam/nonlinear/GncOptimizer.h [hunk 2]
```diff
@@ -291,6 +299,9 @@ class GTSAM_EXPORT GncOptimizer {
                 std::min(mu_init, barcSq_[k] / (2 * rk - barcSq_[k]) ) : mu_init;
           }
         }
+        if (mu_init >= 0 && mu_init < 1e-6)
```
[the remark this answers is not in the document]
Comment (lucacarlone) [reply gtsam/nonlinear/GncOptimizer.h:302]: done!

**Comment by lucacarlone** [comment]
yay! thanks @varunagrawal !

**Pull request merged**